upf/flowtable_init.c

```diff
@@ -65,15 +65,13 @@ flowtable_max_lifetime_update (u16 value)
 }
 
 static clib_error_t *
-flowtable_init_cpu (flowtable_main_t * fm, flowtable_main_per_cpu_t * fmt)
+flowtable_init_cpu (flowtable_main_t * fm, u32 cpu_index)
```

Comment:
I don't see why this change is needed.

Reply:
In debug builds, this function sets `f->cpu_index = cpu_index;` for the flows in the flow entry cache. Before this change, it was erroneously using `os_get_thread_index()` to get that `cpu_index`, which is incorrect b/c this init procedure runs on the main thread, but initializes the per-cpu flowtable structures for all of the cores. Thus, we were getting assertions error later with flows having invalid `cpu_index`. 

Of course, we can keep `flowtable_main_per_cpu_t * fmt` argument and then get proper CPU index like this`u32 cpu_index = fmt - fm->per_cpu`, but I thought that first getting the pointer from the index in `flowtable_init()` and then getting index again from the pointer is a bit excessive.